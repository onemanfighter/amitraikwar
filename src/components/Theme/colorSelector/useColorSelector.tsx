import { useColorMode } from '@chakra-ui/react';

const useColorSelector = () => {
  const { colorMode } = useColorMode();

  return colorMode === 'light'
    ? {
        text: {
          Heading: 'green.500',
          Hero: 'linear(to-b, green.500, blue.500)',
        },
        icon: {
          primary: {
            color: 'green.400',
            bg: 'white',
          },
          secondary: 'green.500',
        },
        bg: {
          container: 'green.200',
        },
        gradient: {
          topAppBar: 'linear(to-r, green.300, blue.500)',
          sideBarBG: 'linear(to-br, green.300, blue.500)',
          contentBG: 'linear(to-r, green.200, blue.400)',
        },
      }
    : {
        text: {
          Heading: 'green.700',
          Hero: 'linear(to-b, green.700, blue.700)',
        },
        icon: {
          primary: {
            color: 'green.700',
            bg: 'white',
          },
          secondary: 'green.700',
        },
        bg: {
          container: 'green.700',
        },
        gradient: {
          topAppBar: 'linear(to-r, green.800, blue.900)',
          sideBarBG: 'linear(to-br, green.800, blue.900)',
          contentBG: 'linear(to-r, green.800, blue.900)',
        },
      };
};

export default useColorSelector;
