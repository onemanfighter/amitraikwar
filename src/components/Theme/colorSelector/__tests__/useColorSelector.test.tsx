import { renderHook } from '@testing-library/react';
import useColorSelector from '../useColorSelector';
import { useColorMode } from '@chakra-ui/react';

jest.mock('@chakra-ui/react', () => ({
  useColorMode: jest.fn(() => ({ colorMode: 'light' })),
}));

const mockUseColorMode = useColorMode as jest.Mock;

describe('useColorSelector', () => {
  it('should return correct color for light mode', () => {
    const { result } = renderHook(() => useColorSelector());

    expect(result.current).toMatchSnapshot();
  });

  it('should return correct color for dark mode', () => {
    mockUseColorMode.mockImplementation(() => ({ colorMode: 'dark' }));
    const { result } = renderHook(() => useColorSelector());

    expect(result.current).toMatchSnapshot();
  });
});
