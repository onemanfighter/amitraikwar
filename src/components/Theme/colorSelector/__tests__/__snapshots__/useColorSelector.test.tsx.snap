// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useColorSelector should return correct color for dark mode 1`] = `
{
  "bg": {
    "container": "green.700",
  },
  "gradient": {
    "contentBG": "linear(to-r, green.800, blue.900)",
    "sideBarBG": "linear(to-br, green.800, blue.900)",
    "topAppBar": "linear(to-r, green.800, blue.900)",
  },
  "icon": {
    "primary": {
      "bg": "white",
      "color": "green.700",
    },
    "secondary": "green.700",
  },
  "text": {
    "Heading": "green.700",
    "Hero": "linear(to-b, green.700, blue.700)",
  },
}
`;

exports[`useColorSelector should return correct color for light mode 1`] = `
{
  "bg": {
    "container": "green.200",
  },
  "gradient": {
    "contentBG": "linear(to-r, green.200, blue.400)",
    "sideBarBG": "linear(to-br, green.300, blue.500)",
    "topAppBar": "linear(to-r, green.300, blue.500)",
  },
  "icon": {
    "primary": {
      "bg": "white",
      "color": "green.400",
    },
    "secondary": "green.500",
  },
  "text": {
    "Heading": "green.500",
    "Hero": "linear(to-b, green.500, blue.500)",
  },
}
`;
